### PR TITLE
properly validate dependencies and settings

### PR DESCRIPTION
**Pull request by xh3b4sd**
Fixes https://github.com/giantswarm/aws-operator/issues/225.

### CODE REVIEW

File: service/create/service.go [hunk 1]
```diff
@@ -73,34 +73,47 @@ func DefaultConfig() Config {
...
+	if config.K8sClient == nil {
+		return nil, microerror.MaskAnyf(invalidConfigError, "config.K8sClient must not be empty")
+	}
 	if config.Logger == nil {
 		return nil, microerror.MaskAnyf(invalidConfigError, "config.Logger must not be empty")
 	}
 
+	// Settings.
+	var emptyAwsConfig awsutil.Config
+	if config.AwsConfig == emptyAwsConfig {
```
Comment (asymmetric) [inline service/create/service.go]: `if config.AwsConfig == awsutil.Config{}`?

Comment (xh3b4sd) [inline service/create/service.go]: https://play.golang.org/p/JWuu7Mavge

Comment (asymmetric) [inline service/create/service.go]: That's pretty weird. This [works](https://play.golang.org/p/e7FxEefnLj) anyway:

    if (config.AwsConfig == awsutil.Config{})

Comment (xh3b4sd) [inline service/create/service.go]: Yeah but then I can just keep it as it is IMO.

**Pull request merged**